### ISSUE TITLE
Switch to test mode during training?

### ISSUE THREAD

**Issue by xuhangc** [tsingqguo/exposure-fusion-shadow-removal#27]
During training process, I encountered the following error message. It seems that param is not in the input variable.

`
switching to testing mode
Traceback (most recent call last):
  File "/home/ipprlab/Downloads/exposure-fusion-shadow-removal/./OE_train.py", line 123, in <module>
    model.set_input(data)
  File "/home/ipprlab/Downloads/exposure-fusion-shadow-removal/models/Fusion_model.py", line 171, in set_input
    self.shadow_param = input['param'].to(self.device).type(torch.float)
KeyError: 'param'
`

**Comment by yuhaoliu7456** [comment]
> During training process, I encountered the following error message. It seems that param is not in the input variable.
> 
> `switching to testing mode Traceback (most recent call last): File "/home/ipprlab/Downloads/exposure-fusion-shadow-removal/./OE_train.py", line 123, in <module> model.set_input(data) File "/home/ipprlab/Downloads/exposure-fusion-shadow-removal/models/Fusion_model.py", line 171, in set_input self.shadow_param = input['param'].to(self.device).type(torch.float) KeyError: 'param'`

During the test, the input['param'] does not exist, you need to write two-line codes to perform the judgement of current state

**Comment by fl82hope** [comment]
yes, here should add some statements like if to check the key exist or not. For the dataset reading, the param doesn't exist in testing.

**Comment by yx-chan131** [comment]
Hi, @xuhangc I encounter the same problem as you. May I know how you fix this problem? 
As @yuhaoliu7456 suggested, I did write an if statement to skip input['param'] during the test. 
```python
if 'param' in input:  #input['param] does not exist during the test
        self.shadow_param = input['param'].to(self.device).type(torch.float)
```
However, I found that variable self.shadow_param will be used in forward method.
```python
addgt = self.shadow_param[:, [0, 2, 4]]
mulgt = self.shadow_param[:, [1, 3, 5]]
```
Therefore I still encounter error when switching to test mode during training.
![error](https://user-images.githubusercontent.com/68681893/139272502-a05e2313-d6f7-4db5-9c4f-7bbf8a7eef95.JPG)
